ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -47,14 +35,36 @@ public partial class PrincipiaPluginAdapter : ScenarioModule {
   private int second_selected_celestial_ = 0;
 
   private bool display_patched_conics_ = false;
-  private bool fix_navball_in_plotting_frame = true;
+  private bool fix_navball_in_plotting_frame_ = true;
 
-  private double[] prediction_steps_ =
-    {1e1, 3e1, 1e2, 3e2, 1e3, 3e3, 1e4, 3e4, 1e5, 3e5, 1e6};
+  // The number of points in a |VectorLine| can be at most 32766, since
+  // Vectrosity imposes a maximum of 65534 vertices, where there are 2 vertices
+  // per point on discrete lines. Moreover there are two points per segment,
+  // so we cannot have 16384 steps (16383 would do).
+  // TODO(egg): At the moment we store points in the history every
+  // 10 n seconds, where n is maximal such that 10 n seconds is less than the
+  // length of a |FixedUpdate|. This means we sometimes have very large gaps.
+  // We should store *all* points of the history, then decimate for rendering.
+  // This means splines are not needed, since 10 s is small enough to give the
+  // illusion of continuity on the scales we are dealing with, and cubics are
+  // rendered by Vectrosity as line segments, so that a cubic rendered as
+  // 10 segments counts 20 towards |kLinePoints| (and probably takes as long to
+  // render as 10 segments from the actual data, with extra overhead for
+  // the evaluation of the cubic).
+  private double[] prediction_step_counts_ =
+      {1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192};
```

Comment:
Could all these constants be written `1 << N`?  That would make it less likely that they contain typos.